### PR TITLE
Making IPPolicy.get_ip_policy_cidrs an inst method

### PR DESCRIPTION
**Pull request by quadewarren**
JIRA:NCP-1315

### CODE REVIEW

File: quark/db/api.py [hunk 1]
```diff
@@ -404,7 +404,10 @@ def ip_address_reallocate_find(context, transaction_id):
 
     # TODO(amir): performance test replacing this with SQL in
     #             ip_address_reallocate's UPDATE statement
-    policy = models.IPPolicy.get_ip_policy_cidrs(subnet)
+    ip_policy_subnet = subnet["ip_policy"]
+    policy = netaddr.IPSet([])
+    if ip_policy_subnet is not None:
+        policy = ip_policy_subnet.get_cidrs_ip_set()
```
Comment (roaet) [inline quark/db/api.py]: For this and the ones below it, should anything happen if the subnet is None?

File: quark/tests/test_db_models.py [hunk 2]
```diff
@@ -1,40 +0,0 @@
...
-        self.assertEqual(ip_policy_rules, IPSet())
-
-    def test_get_ip_policy_cidrs_v6(self):
-        subnet = dict(id=1, ip_version=6, next_auto_assign_ip=0,
-                      cidr="fc00::/7",
-                      first_ip=334965454937798799971759379190646833152L,
-                      last_ip=337623910929368631717566993311207522303L,
-                      network=dict(ip_policy=None), ip_policy=None)
-        ip_policy_rules = models.IPPolicy.get_ip_policy_cidrs(subnet)
-        self.assertEqual(ip_policy_rules, IPSet())
```
Comment (roaet) [inline quark/tests/test_db_models.py]: Can't believe there was a whole file dedicated to testing the jank

File: quark/db/api.py [hunk 3]
```diff
@@ -404,7 +404,10 @@ def ip_address_reallocate_find(context, transaction_id):
 
     # TODO(amir): performance test replacing this with SQL in
     #             ip_address_reallocate's UPDATE statement
-    policy = models.IPPolicy.get_ip_policy_cidrs(subnet)
+    ip_policy_subnet = subnet["ip_policy"]
+    policy = netaddr.IPSet([])
```
Comment (xroot88) [inline quark/db/api.py]: I think line 408 should go into the else part of the if for readability:
```python
if ip_policy_subnet is not None:
    policy = ip_policy_subnet.get_cidrs_ip_set()
else:
    policy = netaddr.IPSet([])
```

File: quark/db/models.py [hunk 4]
```diff
@@ -327,7 +327,10 @@ def allocation_pools(self):
             pools = json.loads(_cache)
             return pools
         else:
-            ip_policy_cidrs = IPPolicy.get_ip_policy_cidrs(self)
+            ip_policy = self["ip_policy"]
```
Comment (xroot88) [inline quark/db/models.py]: same as in prev comment

File: quark/ipam.py [hunk 5]
```diff
@@ -452,7 +452,10 @@ def _allocate_from_subnet(self, context, net_id, subnet,
                                                  port_id=port_id,
                                                  ip_address=ip_address)))
 
-        ip_policy_cidrs = models.IPPolicy.get_ip_policy_cidrs(subnet)
+        ip_policy_subnet = subnet["ip_policy"]
```
Comment (xroot88) [inline quark/ipam.py]: same as above

File: quark/db/api.py [hunk 6]
```diff
@@ -404,7 +404,15 @@ def ip_address_reallocate_find(context, transaction_id):
 
     # TODO(amir): performance test replacing this with SQL in
     #             ip_address_reallocate's UPDATE statement
-    policy = models.IPPolicy.get_ip_policy_cidrs(subnet)
+    ip_policy_subnet = None
```
Comment (xroot88) [inline quark/db/api.py]: It looks like this fragment is repeated 4 times - just wondering why not make it a function and put it in some common file?

Comment (quadewarren) [inline quark/db/api.py]: That is actually what we were replacing. Originally, this was a function attached to the IPPolicy class, but we made it an instance method, so these checks became needed. We could wrap it in a function, but I don't think this snippet of code belongs in models.py and if we created a new file to contain it, we might as well shove other common functions, too. I don't think that would make the code work more clearly, however.

**Comment by roaet** [comment]
LGTM +1

🐣

File: quark/ipam.py [hunk 7]
```diff
@@ -518,7 +526,15 @@ def _allocate_from_v6_subnet(self, context, net_id, subnet,
...
-            ip_policy_cidrs = models.IPPolicy.get_ip_policy_cidrs(subnet)
+            ip_policy_subnet = None
+            if subnet is not None:
+                ip_policy_subnet = subnet["ip_policy"]
+
+            if ip_policy_subnet is not None:
+                ip_policy_cidrs = ip_policy_subnet.get_cidrs_ip_set()
+            else:
+                ip_policy_cidrs = netaddr.IPSet([])
+
```
Comment (xroot88) [inline quark/ipam.py]: Maybe shorten this block a little to something like this?:
```python
            if subnet and subnet['ip_policy']:
                ip_policy_cidrs = subnet['ip_policy'].get_cidrs_ip_set()
            else:
                ip_policy_cidrs = netaddr.IPSet([])
```

**Comment by xroot88** [comment]
+1 - check my comment to make code more readable, otherwise, looks good

File: quark/ipam.py (hunk 7 above)
Comment (quadewarren) [inline quark/ipam.py]: That works.

**Pull request merged**